tests/integration/backward_compatible/sql_test.py

```diff
@@ -0,0 +1,590 @@
+        statement.timeout = 100
+        result = self.client.sql.execute_statement(statement)
+
+        six.assertCountEqual(
+            self, [i for i in range(10)], [row.get_object("age") for row in result]
+        )
+
+    def test_execute_statement_with_cursor_buffer_size(self):
+        entry_count = 50
+        self._populate_map(entry_count, lambda v: Student(v, v))
+        statement = SqlStatement("SELECT age FROM %s" % self.map_name)
+        statement.cursor_buffer_size = 3
+        result = self.client.sql.execute_statement(statement)
+
+        six.assertCountEqual(
+            self, [i for i in range(entry_count)], [row.get_object("age") for row in result]
+        )
+
+    def test_execute_statement_with_copy(self):
+        self._populate_map()
+        statement = SqlStatement("SELECT __key FROM %s WHERE this >= ?" % self.map_name)
+        statement.parameters = [9]
+        copy_statement = statement.copy()
+        statement.clear_parameters()
+
+        result = self.client.sql.execute_statement(copy_statement)
+        self.assertEqual([9], [row.get_object_with_index(0) for row in result])
+
+        result = self.client.sql.execute_statement(statement)
+        with self.assertRaises(HazelcastSqlError):
+            for _ in result:
+                pass
+
+    # Can't test the case we would expect an update count, because the IMDG SQL
+    # engine does not support such query as of now.
+    def test_execute_statement_with_expected_result_type_of_rows_when_rows_are_expected(self):
+        entry_count = 100
+        self._populate_map(entry_count, lambda v: Student(v, v))
+        statement = SqlStatement("SELECT age FROM %s WHERE age < 3" % self.map_name)
+        statement.expected_result_type = SqlExpectedResultType.ROWS
+        result = self.client.sql.execute_statement(statement)
+
+        six.assertCountEqual(self, [i for i in range(3)], [row.get_object("age") for row in result])
+
+    # Can't test the case we would expect an update count, because the IMDG SQL
+    # engine does not support such query as of now.
+    def test_execute_statement_with_expected_result_type_of_update_count_when_rows_are_expected(
+        self,
+    ):
+        self._populate_map()
+        statement = SqlStatement("SELECT * FROM %s" % self.map_name)
+        statement.expected_result_type = SqlExpectedResultType.UPDATE_COUNT
+        result = self.client.sql.execute_statement(statement)
+
+        with self.assertRaises(HazelcastSqlError):
+            for _ in result:
+                pass
+
+    # Can't test the schema, because the IMDG SQL engine does not support
+    # specifying a schema yet.
+
+
+class SqlResultTest(SqlTestBase):
+    def test_blocking_iterator(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT __key FROM %s" % self.map_name)
+
+        six.assertCountEqual(
+            self, [i for i in range(10)], [row.get_object_with_index(0) for row in result]
+        )
```

Comment:
This is already tested several times, do we need this one?

Reply:
To make the tests symmetrical. I would like to have a standard way of testing things even though this requires a little bit of duplication.